tools/unit_tests.ts

```diff
@@ -0,0 +1,119 @@
+import { args, exit, run } from "deno";
+
+function extractNumber(pattern: RegExp, str: string) {
+  const matches = pattern.exec(str);
+  if (matches == null || pattern.exec(str) != null) {
+    return;
+  }
+  return Number(matches[1]);
+}
+
+function parseUnitTestOutput(
+  output: string
+): [string | undefined, string | undefined] {
+  let expected;
+  let actual;
+  let result;
+  for (const line of output.split("\n")) {
+    if (expected == null) {
+      // expect "running 30 tests"
+      expected = extractNumber(/running (\d+) tests/g, line);
+    } else if (line.includes("test result:")) {
+      result = line;
+    }
+    console.log(line);
+  }
+  // Check that the number of expected tests equals what was reported at the
+  // bottom.
+  if (result) {
+    // result should be a string like this:
+    // "test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; ..."
+    actual = extractNumber(/(\d+) passed/g, result);
+  }
+  return [actual, expected];
+}
+
+interface ReadResult {
+  eof: boolean;
+  nread: number;
+}
+
+interface Reader {
+  read(p: Uint8Array): Promise<ReadResult>;
+}
+
+async function readAll(reader: Reader) {
+  let result: Uint8Array;
+  while (true) {
+    const buffer = new Uint8Array(4096);
+    const { eof, nread } = await reader.read(buffer);
+    if (result) {
+      const tmp = result;
+      result = new Uint8Array(tmp.length + nread);
+      result.set(tmp);
+      result.set(buffer.slice(0, nread), tmp.length);
+    } else {
+      result = buffer.slice(0, nread);
+    }
+    if (eof) {
+      break;
+    }
+  }
+  return new TextDecoder("utf-8").decode(result);
+}
```

Comment:
This utility is very useful to have, and I've meant to add it before. I'd rather if we did this in a separate PR tho, and if you put it into `js/io.ts` ... It should use the `deno.Buffer` class, which provides most of the functionality. 

Here's an example: https://github.com/denoland/deno/blob/2ffd78daf9956a24098d1f959f21882e350e9d37/js/fetch.ts#L25-L28

Would you like to do this separate PR for `deno.readAll(r: Reader): Uint8Array` ? 
If so I would ask you look at the implementation in Golang and try to copy it as closely as possible: https://github.com/golang/go/blob/048c9164a0c5572df18325e377473e7893dbfb07/src/io/ioutil/ioutil.go#L16-L46
including this test:
https://github.com/golang/go/blob/048c9164a0c5572df18325e377473e7893dbfb07/src/io/ioutil/example_test.go#L16-L28